Library/src/Multiplayer/SpaceEntitySystem.cpp

```diff
@@ -849,6 +849,21 @@ void SpaceEntitySystem::OnAllEntitiesCreated()
 	// Ensure entity list is up to date
 	ProcessPendingEntityOperations();
 
+	// Resolve entity hierarchy
+	for (size_t i = 0; i < Entities.Size(); ++i)
+	{
+		SpaceEntity* Entity = Entities[i];
+
+		if (Entity->ParentId.HasValue())
```

Comment:
Any reason to not invoke the new `ResolveEntityHierarchy` function? Could be an opportunity to follow the DRY pattern.

Reply:
ResolveEntityHierarchy relies on ShouldUpdateParent being set, which isn't the case in this circumstance. I could manually set this on deserialisation if we want to go down this route, but I think we should consider the following:

. While I understand it removes some repeated code, ResolveEntityHierarchy also contains logic that isn't necessary in this case (removing the parent and removing previous parents), which may make it harder to understand/debug.

Reply:
Makes sense. Thanks for the explanation!